packages/lastrev-build-prefetch-content/src/buildTasks/writeWebsiteSectionPaths.ts

```diff
@@ -0,0 +1,124 @@
+import { getAllContentItemsForContentType } from '@last-rev/integration-contentful';
+import { each, get, set, map } from 'lodash';
+import { CONTENT_DIR, PATHS_FILE } from '../constants';
+import writeFile from '../helpers/writeFile';
+import mkdirIfNotExists from '../helpers/mkDirIfNotExists';
+import { BuildTask } from '../types';
+
+const hashResults = (results) => {
+  const out = {};
+  each(results, (result) => {
+    const id = get(result, 'sys.id');
+    if (!id) return;
+
+    out[id] = result;
+  });
+
+  return out;
+};
+
+const getDetailPagePaths = (paths, pages, websiteSections) => {
+  each(pages, (page, contentId) => {
+    const parents = get(page, 'fields.targeting');
+    const contentTypeId = get(page, 'sys.contentType.sys.id');
+    each(parents, (parent) => {
+      let parentId = get(parent, 'sys.id');
+      if (!parentId) return;
+
+      let wss;
+      let componentId;
+      let path = get(page, 'fields.slug');
+      let domain;
+      while (!domain) {
```

Comment:
In my experience it's a little hard to understand which variables are strictly required for each level of a while loop.
To make this more maintainable we could try using a more functional approach:

```js
    const domain = parents
      // Find parentId and filter out falsy values
      .map(prop("sys.id"))
      .filter(identity)

      // Find the domain
      .find((parentId) => {
        const wss = get(websiteSections, parentId);
        if (!wss) return false;

        ... rest of logic
      });
```

This would make it easier for the next reader to determine what's happening at each level of the loop.